nifi-extension-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/GetSlackReaction.java

```diff
@@ -0,0 +1,294 @@
+            .build();
+
+    static final Relationship REL_WAIT = new Relationship.Builder()
+            .name("wait")
+            .description("Self loop relationship which is used when waiting for further reactions")
+            .build();
+
+    static final Relationship REL_NO_REACTION = new Relationship.Builder()
+            .name("no_reaction")
+            .description("FlowFiles are routed to this relationship if no reaction arrived for the message in the given timeframe")
+            .build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return Arrays.asList(CHANNEL_ID,
+                             ACCESS_TOKEN,
+                             THREAD_TIMESTAMP,
+                             WAIT_MONITOR_WINDOW,
+                             RELEASE_IF_ONE_REACTION);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return Set.of(REL_SUCCESS,
+                REL_WAIT,
+                REL_NO_REACTION,
+                REL_FAILURE);
+    }
+
+    public static final String ATTR_WAIT_TIME = "minutes.waited";
+    public static final String ATTR_ERROR_MESSAGE = "error.message";
+    public static final String ATTR_MESSAGE_TIMESTAMP = "slack.message.timestamp";
+    public static final String ATTR_CHANNEL_ID = "slack.channel.id";
+    public static final int PENALTY_MINUTES = 5;
+
+    @Override
+    public boolean isStateful(final ProcessContext context) {
+        return true;
+    }
+
+    private RateLimit rateLimit;
+    private volatile App slackApp;
+    private MethodsClient client;
+
+    @OnScheduled
+    public void setup(final ProcessContext context) {
+        rateLimit = new RateLimit(getLogger());
+        slackApp = createSlackApp(context);
+        client = initializeClient(slackApp);
+    }
+
+    @OnStopped
+    public void shutdown() {
+        if (slackApp != null) {
+            slackApp.stop();
+            slackApp = null;
+        }
+        rateLimit = null;
+    }
+
+    public RateLimit getRateLimit() {
+        return rateLimit;
+    }
+
+    private App createSlackApp(final ProcessContext context) {
+        final String botToken = context.getProperty(ACCESS_TOKEN).getValue();
+        final AppConfig appConfig = AppConfig.builder()
+                .singleTeamBotToken(botToken)
+                .build();
+
+        return new App(appConfig);
+    }
+
+    protected MethodsClient initializeClient(final App slackApp) {
+        slackApp.start();
+        return slackApp.client();
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        if (rateLimit.isLimitReached()) {
+            getLogger().debug("Will not fetch reactions from Slack because rate limit has been reached");
+            context.yield();
+            return;
+        }
+
+        FlowFile flowFile = session.get();
```

Comment:
Most code in the code base has logic if the `flowFile` is null.
```suggestion
        FlowFile flowFile = session.get();
        if (flowFile == null) {
            return;
        }
```